cranelift/filetests/src/test_zkasm.rs

```diff
@@ -237,6 +240,62 @@ mod tests {
         program.join("\n")
     }
 
+    fn run_wat_file(path: &Path) -> Result<(), Box<dyn std::error::Error>> {
+        let engine = Engine::default();
+        let wat = std::fs::read_to_string(path).unwrap();
+        let pat = r#"(import "env" "assert_eq" (func $assert_eq (param i32) (param i32)))"#;
+        let assert_type = if wat.contains(pat) { 32 } else { 64 };
```

Comment:
This seems like an awfully brittle way to check for this. If somebody puts any other whitespace, or just names $name differently suddenly the entire thing breaks.

I was about to suggest to properly parse the wasm, but there is really an easier way to do this. Use two different host functions (think: `assert_eq_32` and `assert_eq_64`) and implement them both unconditionally here.

Reply:
It seems like it is matching this imports with ``(import ...)`` statements in ``.wat`` file, so we can't simply add second function. I guess here it will be better to introduce some more robust ``.wat`` parsing

Reply:
> It seems like it is matching this imports with (import ...) statements in .wat file, so we can't simply add second function

It works in the other direction – the `(import ...)` statements match against the list of provided host functions (such as here,) to select ones to link against. It is perfectly fine if the host provides more functions than what `wat` requires to execute – those can be simply ignored.

The only thing you need to ensure here is that the names for the different functions are… different. Which they arguably should be if the function signatures are different.

Yes, you will need to adjust the generated spec tests, but in the long run it is better that this is done earlier rather than later. Otherwise this codebase will continue accumulating hacks like these all over the place, rather than doing the simple thing.

Reply:
Hm, maybe you know way to provide an extra host functions?
I'm trying something like:
```
        let mut store = Store::new(&engine, ());
        let assert_eq_64 = Func::wrap(&mut store, |a: i64, b: i64| -> () {
            assert_eq!(a, b);
        });
        let asser_eq_32 = Func::wrap(&mut store, |a: i32, b: i32| -> () {
            assert_eq!(a, b);
        });
        let _instance = Instance::new(
            &mut store,
            &module,
            &[asser_eq_32.into(), assert_eq_64.into()],
        )?;
 ```
 But it fails with
 ```
 Err = Err(
    "expected 1 imports, found 2",
)
```